packages/server/lib/request.js

```diff
@@ -698,6 +698,11 @@ module.exports = function (options = {}) {
       // either turn these both on or off
       options.followAllRedirects = options.followRedirect
 
+      // https://github.com/cypress-io/cypress/issues/28789
+      if (options.json === true) {
+        if (options.body === true || options.body === false || options.body === null) options.body = String(options.body)
```

Comment:
This might read a bit better and grab anything that's built with the constructor but shouldn't make a difference either way
```suggestion
        if (_.isBoolean(options.body) || _.isNull(options.body)) options.body = String(options.body)
```